lib/backend/firestore/firestorebk.go

```diff
@@ -709,19 +709,38 @@ func (b *Backend) purgeExpiredDocuments() error {
 // deleteDocuments removes documents from firestore in batches to stay within the
 // firestore write limits
 func (b *Backend) deleteDocuments(docs []*firestore.DocumentSnapshot) error {
-	for i := 0; i < len(docs); i += commitLimit {
-		batch := b.svc.Batch()
+	mDocs := make(map[string]*firestore.DocumentSnapshot)
+	for _, doc := range docs {
+		mDocs[doc.Ref.Path] = doc
+	}
```

Comment:
This section doesn't seem to be used except for `jobs := make([]*firestore.BulkWriterJob, 0, len(mDocs))`.

Probably you wanted to use the `mDocs` in the for loop instead of docs?

Reply:
Huh, do we have a review race condition here perhaps? I did update the loop in that commit to user mDocs. In https://github.com/gravitational/teleport/pull/21190/commits/e7b5f0d17975d8dd464494f8cb9bffa999a9bcdc

Reply:
I don't think we need to consider any race condition here.
Anw, a final nit to avoid ranging a map 

```go
// deleteDocuments removes documents from firestore in batches to stay within the
// firestore write limits
func (b *Backend) deleteDocuments(docs []*firestore.DocumentSnapshot) error {
	set := make(map[string]struct{})

	batch := b.svc.BulkWriter(b.clientContext)
	jobs := make([]*firestore.BulkWriterJob, 0, len(docs))

	i := 0
	for _, doc := range docs {
		// Deduplicate documents. The Firestore SDK will error if duplicates are found,
		// but existing callers of this functionassume this is valid.
		if _, ok := set[doc.Ref.Path]; ok {
			continue
		}
		set[doc.Ref.Path] = struct{}{}

		job, err := batch.Delete(doc.Ref)
		if err != nil {
			return ConvertGRPCError(err)
		}

		jobs = append(jobs, job)
		i++
		if i >= commitLimit {
			batch.Flush()
		}
	}
       batch.End()
	var errs []error
	for _, job := range jobs {
		if _, err := job.Results(); err != nil {
			errs = append(errs, ConvertGRPCError(err))
		}
	}

	return trace.NewAggregate(errs...)
}
```

Reply:
Updated, but is there anything bad about ranging a map?